com.avaloq.tools.ddk.xtext.valid.generator/src/com/avaloq/tools/ddk/xtext/valid/generator/ValidValidatorFragment.java

```diff
@@ -310,8 +286,7 @@ private static String getPreferenceKey(final Rule rule) {
    * @return the preference key
    */
   public static String getPreferenceKey(final Grammar grammar, final Rule rule) {

```

Comment:
I think this is only used in JavaValidators.xpt (via ValidExtensions.ext) to check the enablement state (in the preferences) of a validation. Assuming that is the only place this method is used, I think it can also be removed and JavaValidators.xpt can be simplified.

Reply:
Hi, Is it possible to have rules within rules, or categories within categories? 
If so, is it possible to define recursive functions in .xpt?
It looks like there are two helper functions in ValidValidatorFragment that could possibly be recursive calls?
Not quite sure.

Reply:
Thanks for the chat, Andrew. I removed this one and made some changes to JavaValidators.xpt.